src/extension.ts

```diff
@@ -65,10 +65,17 @@ export async function activate(context: ExtensionContext) {
 }
 
 function fetchAndLaunchMetals(context: ExtensionContext, javaHome: string) {
-  if (fs.existsSync(dottyIdeArtifact())) {
+  if (!workspace.workspaceFolders) {
+    outputChannel.appendLine(
+      `Metals will not start because you've opened a single file and not a project directory.`
```

Comment:
👍 Maybe we can do better in the future but this is already a big improvement over crashing